client/src/components/DataFiles/DataFilesProjectMembers/_cells/ProjectRoleSelector.jsx

```diff
@@ -0,0 +1,86 @@
+import React, { useState, useEffect, useContext, createContext } from 'react';
+import { useQuery, useMutation, useQueryClient } from 'react-query';
+import Cookies from 'js-cookie';
+import fetch from 'cross-fetch';
+import DropdownSelector from '_common/DropdownSelector';
+import { Button } from 'reactstrap';
+import LoadingSpinner from '_common/LoadingSpinner';
+
+const getProjectRole = async (projectId, username) => {
+  const url = `/api/projects/${projectId}/project-role/${username}/`;
+  const request = await fetch(url, {
+    headers: { 'X-CSRFToken': Cookies.get('csrftoken') },
+    credentials: 'same-origin',
+  });
+  const data = await request.json();
+  return data;
+};
+
+const setProjectRole = async (projectId, username, oldRole, newRole) => {
+  const url = `/api/projects/${projectId}/members/`;
+  const request = await fetch(url, {
+    headers: { 'X-CSRFToken': Cookies.get('csrftoken') },
+    credentials: 'same-origin',
+    method: 'PATCH',
+    body: JSON.stringify({
+      action: 'change_project_role',
+      username,
+      oldRole,
+      newRole,
+    }),
+  });
+  const data = await request.json();
+  return data;
+};
+
+const useProjectRole = (projectId, username) => {
```

Comment:
Should we have hooks in separate files for visibility?

Reply:
Yeah, we should. It should live in the hooks module though, so I think the hooks can stay here until we merge in the PR with that module.